embeddings-transformers-noc.js

```diff
@@ -0,0 +1,79 @@
+import * as fs from 'fs';
+import { pipeline } from '@xenova/transformers';
+
+// Load the embeddings model
+const extractor = await pipeline('feature-extraction', 'Xenova/bge-small-en-v1.5');
+
+const fullOutput = [];
+
+(async () => {
+  // Scan transcripts directory for all json files
+  const files = fs.readdirSync('transcripts/markdown');
+
+  // Iterate through each file and calculate the embeddings
+  for (const file of files) {
+    const text = fs.readFileSync(`transcripts/markdown/${file}`, 'utf-8');
+    // const json = JSON.parse(rawContents);
+
+    // Calculate chunks based on this text
+    const chunks = calculateMarkdownChunks(text);
+
+    // Extract embeddings for each chunk
+    const output = [];
+
+    for (const chunk of chunks) {
+      const embeddingOutput = await extractor(chunk, {
+        pooling: 'mean',
+        normalize: true,
+      });
+
+      const embedding = embeddingOutput.tolist()[0];
+      output.push({ text: chunk, embedding });
+      fullOutput.push({ text: chunk, embedding });
+    }
+
+    // Save the embeddings to a file
+    const fileOut = `embeddings/${file}`;
+    fs.writeFileSync(fileOut, JSON.stringify(output));
+
+    console.log(
+      `Embeddings saved for ${file} to ${fileOut} (${output.length} chunks) (${
+        files.indexOf(file) + 1
+      }/${files.length})`
+    );
```

Comment:
not that it matters, but these embeddings are being saved in `.md` files instead of `.json` files.

Reply:
oh whoops, yes, that was just a mistake!